### DISCUSSION TITLE
does this work with hermes?

### DISCUSSION THREAD

**Discussion by ethanneff** [invertase/react-native-firebase#6559] [Q&A]
If not, is support for it actively being worked on?

**Comment by mikehardy** [comment]
Yes:
https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh

Careful of a couple finer points:

- https://github.com/mikehardy/rnfbdemo/blob/7ab856ae3befcd431abf93f57481ed51f9658945/make-demo.sh#L98-L99
- https://github.com/mikehardy/rnfbdemo/blob/7ab856ae3befcd431abf93f57481ed51f9658945/make-demo.sh#L105-L107

It was quite the struggle, but it should work.
- https://github.com/reactwg/react-native-releases/discussions/21#discussioncomment-2922651
- https://github.com/reactwg/react-native-releases/discussions/26#discussioncomment-3441178
- some others...

Anyway, it is *supposed* to work now. The script I linked exercises it all the time, it's my test rig to fuzzbust through these things. Please let me + react-native folks know if it does not.

Flipper is still a no, new architecture is still a no.

**Reply by tyrauber** [reply]
Hey @mikehardy, thanks for this. 

I'd been struggling to get react-native-firebase working in an existing Expo app.  No matter what I did, it would work locally, but when I cut an internal or store build it would crash on launch in iOS.

After a lot of trial and error,  I created an example repo, and ultimately determined the issue was Hermes on iOS.

As far as I can tell, Hermes does not work with react-native-firebase on 0.69.x and Expo SDK 46.  Here is an [working example repo](https://github.com/tyrauber/expo-firebase-example) without Hermes. If you add `jsEngine: "hermes"` to app.json, the same code base crashes on iOS.

This only happens on production. It works fine locally, unfortunately, but internal or store builds crash on launch. Android works fine.

package.json
```
 "expo": "~46.0.16",
 "expo-dev-client": "~1.3.1",
 "expo-build-properties": "~0.3.0",
 "@react-native-firebase/analytics": "^16.2.0",
 "@react-native-firebase/app": "~16.2.0",
 "@react-native-firebase/crashlytics": "^16.2.0",
 "@react-native-firebase/perf": "^16.2.0",
 "react": "18.0.0",
 "react-dom": "18.0.0",
 "react-native": "0.69.6",
 ```
 
app.json
```
    "plugins": [
      [
        "expo-build-properties",
        {
          "ios": {
            "useFrameworks": "static"
          }
        }
      ],
      "@react-native-firebase/app",
      "@react-native-firebase/crashlytics",
      "@react-native-firebase/perf"
    ]
  ```

 @brentvatne, Any idea why `react-native-firebase` works locally on iOS with Hermes, but iOS EAS builds crash on launch?

**Reply by mikehardy** [reply]
Do you have a crashtrace (perhaps from `@react-native-firebase/crashlytics` - I can see it is installed)

**Reply by tyrauber** [reply]
Hi @mikehardy, When Hermes is enabled, the crash happens immediately on app launch. I don't think the crash gets reported to Crashlytics.  If I look at the last 60 mins in the dashboard, I see 0 crashes, and I've launched and crashed the app a dozen times.  When Hermes is disabled, I can manually crash the app and the crash is reported to the dashboard.

Note: I do see a red banner now at the top of Crashlaytics that says:
> This app has 4 unprocessed crashes. Upload 1 dSYM file to process them.

Perhaps related?  Happy to keep trying stuff. Would love to help get 16.2.0 working with Expo 46, RN 0.69.x and Hermes. The repo is pretty straightforward and quite easy to build and test on EAS.  

Thanks for the time. Appreciate you building and maintaining this library.

**Reply by mikehardy** [reply]
Plug device into computer, open console.app, select the device on left side, trigger crash, grab stack trace

You should definitely upload your dsyms

**Reply by tyrauber** [reply]
Ah, Got it, thanks. This seems to be the cause right here:
```
*** Terminating app due to uncaught exception 'RCTFatalException: Could not get BatchedBridge, make sure your bundle is packaged correctly', reason: 'Could not get BatchedBridge, make sure your bundle is packaged correctly'
*** First throw call stack:
(0x1d71ad288 0x1efedd744 0x10471f5f0 0x1047372bc 0x1047357f8 0x10474a0c4 0x104749e58 0x1d71865a4 0x1d7187500 0x1d711f048 0x1d7132bc8 0x104733078 0x1d897541c 0x2483699ac 0x248368e68)
```
But I can post the whole stack trace if you want it.

**Reply by mikehardy** [reply]
Seems sufficient. Either the bundle was not compiled + included + archived correctly or Hermes is not able to find it for some reason. Not a react-native-firebase thing I don't think. You might try same with hermes+frameworks + app-with-no-other-modules
